File: app/src/main/java/com/fenger/fragmentdemo/Permission/PermissionFail.java
package com.fenger.fragmentdemo.Permission;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * com.fenger.permission
 * Created by fenger
 * in 2019-12-12
 *
 * @author kissgakki
 */

@Target(ElementType.METHOD)//放在"方法"上面
@Retention(RetentionPolicy.RUNTIME)//编译时还是运行时监测？
public @interface PermissionFail {
    public int requestCode();
}

File: app/src/main/java/com/fenger/fragmentdemo/permission/PermissionFail.kt
package com.fenger.fragmentdemo.permission

/**
 * com.fenger.permission
 * Created by fenger
 * in 2019-12-12
 *
 * @author kissgakki
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER) //放在"方法"上面
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME) //编译时还是运行时监测？
annotation class PermissionFail(val requestCode: Int)
